config/jobs/kubernetes-sigs/cluster-api/cluster-api-ci.yaml

```diff
@@ -24,6 +24,7 @@ periodics:
     testgrid-num-failures-to-alert: "2"
 - name: periodic-cluster-api-e2e-test
   interval: 1h
+  run_if_changed: '^(docs/|.*\.md$)'
```

Comment:
this would do the reverse, it would run the tests **only** on /docs and *.md changes.
see here:
https://github.com/kubernetes/test-infra/issues/13032#issuecomment-501953200

Reply:
That's correct. I misread the initial `^` as a negation whereas in this case it indicates the start of the line.

Reply:
We don't have to do this today, we can optimize these down the line — There isn't really a good way to do this because go regexp (used here) doesn't support negative lookahead

Reply:
due to the language limitations i can see e.g. `skip_if_changed` that takes precedence and errors out of both are in the prow config. but i vaguely recall arguments against this feature. #prow on k8s slack should know more.

Reply:
If we want to implement only running on docs/md changes, we'd probably need to model after the CAPV configuration, which explicitly lists paths (go packages + files like main.go, go.mod/sum, etc) for code changes for unit/e2e jobs. This becomes a maintenance burden, however, because it needs to be kept up to date if/when we add new packages or files that affect compilation.

Reply:
Will create another PR for this.

Reply:
https://github.com/kubernetes/test-infra/pull/19103